query-coordinator/src/main/scala/com/socrata/querycoordinator/rollups/QueryRewriterImplementation.scala

```diff
@@ -225,7 +225,7 @@ class QueryRewriterImplementation(analyzer: SoQLAnalyzer[SoQLType, SoQLValue]) e
         else h
       }
 
-      val selection = rewriteSelection(q.selection, r, rollupColIdx,(where.isDefined && groupBy.nonEmpty && having.isDefined)) map { s =>
```

Comment:
Instead of judging "do all clauses match?" by a "successful rewrite", lets instead do a literal expression equality check.